File: packages/urdu-js/src/normalization/diacritics.ts
import { DIACRITIC_RANGES } from './unicode-data.js'
import { stripRanges } from './normalize.js'

export function stripDiacritics(text: string): string {
  return stripRanges(text, DIACRITIC_RANGES)
}

export function isDiacritic(codePoint: number): boolean {
  return DIACRITIC_RANGES.some(([lo, hi]) => codePoint >= lo && codePoint <= hi)
}

File: packages/urdu-dart/lib/src/normalization/diacritics.dart
import 'unicode_data.dart';
import 'normalize.dart';

/// Removes all Urdu/Arabic diacritics from text.
String stripDiacritics(String text) => stripRanges(text, diacriticRanges);

/// Returns true if the given code point is a diacritic.
bool isDiacritic(int codePoint) =>
    diacriticRanges.any((r) => codePoint >= r.$1 && codePoint <= r.$2);
